segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -639,10 +640,9 @@ private int readInternal(SegmentHandle handle, byte[] buffer, long offset, int b
         //There is only one file per segment.
         FileStatus currentFile = findStatusForSegment(handle.getSegmentName(), true);
         try (FSDataInputStream stream = this.fileSystem.open(currentFile.getPath())) {
-            if (offset + length > stream.available()) {
-                throw new ArrayIndexOutOfBoundsException();
-            }
             stream.readFully(offset, buffer, bufferOffset, length);
+        } catch (EOFException e) {
```

Comment:
`readInternal` is called from `read`. if we throw here, then this exception is going to bubble up. we should be unit testing this behavior of the `read` call for the case that the offset is beyond the end of the file.

Reply:
This case covers this scenario: https://github.com/pravega/pravega/blob/76124d152ec77510970006d1b6b4d1a4a7a317c6/segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/StorageTestBase.java#L216